nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -562,11 +562,9 @@ sub addProfile {
             next;
         }
         my $date = strftime("%F", localtime(lstat($link)->mtime));
-        my $version =
-            -e "$link/nixos-version"
-            ? readFile("$link/nixos-version")
-            : basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
-        addEntry("NixOS - Configuration " . nrFromGen($link) . " ($date - $version)", $link);
+        my $nixos_version = readFile("$link/nixos-version");
+        my $kernel_version = basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
+        addEntry("NixOS $nixos_version - generation " . nrFromGen($link) . ", kernel $kernel_version, built $date", $link);
```

Comment:
```suggestion
        addEntry("NixOS $nixos_version - generation " . nrFromGen($link) . ", kernel $kernel_version, added $date", $link);
```

All we know about the build is that it happened in the past, probably. :alien: